src/mammos_ai/beyond_stoner_wohlfarth_fixed_angle/bsw.py

```diff
@@ -0,0 +1,176 @@
+    model: str = "random-forest-v1",
+) -> str | list[str]:
+    """Classify material as soft or hard magnetic from micromagnetic properties.
+
+    This function classifies a magnetic material as either soft or hard magnetic
+    based on its micromagnetic properties spontaneous magnetization Ms, exchange
+    stiffness constant A and uniaxial anisotropy constant Ku.
+
+    Args:
+       Ms: Spontaneous magnetization.
+       A: Exchange stiffness constant.
+       Ku: Uniaxial anisotropy constant.
+       model: AI model used for the classification
+
+    Returns:
+       Classification as "soft" or "hard".
+
+    """
+    Ms = me.Ms(Ms, unit=u.A / u.m)
+    A = me.A(A, unit=u.J / u.m)
+    Ku = me.Ku(Ku, unit=u.J / u.m**3)
+
+    Ms_arr = np.atleast_1d(Ms.value)
+    A_arr = np.atleast_1d(A.value)
+    Ku_arr = np.atleast_1d(Ku.value)
+
+    if not (Ms_arr.shape == A_arr.shape == Ku_arr.shape):
+        raise ValueError(
+            f"Input arrays must have the same length. Shapes are Ms: {Ms_arr.shape}, "
+            f"A: {A_arr.shape}, Ku: {Ku_arr.shape}"
+        )
+
+    match model:
+        case "random-forest-v1":
+            classifier_path = _MODEL_DIR / "classifier_random_forest_v1.onnx"
+        case _:
+            raise ValueError(f"Unknown model {model}")
+
+    session = ort.InferenceSession(str(classifier_path), _SESSION_OPTIONS)
+    X = np.column_stack([Ms_arr, A_arr, Ku_arr]).astype(np.float32)
+
+    results = session.run(None, {session.get_inputs()[0].name: X})[0]
+    labels = np.where(results == 0, "soft", "hard")
+
+    if X.shape[0] == 1:
+        return labels.item()
+    return labels.flatten().tolist()
+
+
+def Hc_Mr_BHmax_from_Ms_A_K(
+    Ms: mammos_entity.Entity | astropy.units.Quantity | numbers.Number,
+    A: mammos_entity.Entity | astropy.units.Quantity | numbers.Number,
+    Ku: mammos_entity.Entity | astropy.units.Quantity | numbers.Number,
+    model: str = "random-forest-v1",
+) -> mammos_analysis.hysteresis.ExtrinsicProperties:
+    """Predict Hc, Mr and BHmax from micromagnetic properties Ms, A and Ku.
+
+    This function predicts extrinsic properties coercive field Hc, remanent
+    magnetization Mr and maximum energy product BHmax given a set of micromagnetic
+    material parameters.
+
+    Different models are available for the prediction. The following list provides an
+    overview:
+
+    - ``random-forest-v1``: TODO explain details of model, training data,
+      Zenodo DOI to dataset and simulation script
+
+
+    Args:
+       Ms: Spontaneous magnetization.
+       A: Exchange stiffness constant.
+       Ku: Uniaxial anisotropy constant.
+       model: AI model used for the prediction
+
+    Returns:
+       Extrinsic properties Hc, Mr, BHmax
+
+    Examples:
+    >>> import mammos_ai
+    >>> import mammos_entity as me
+    >>> mammos_ai.Hc_Mr_BHmax_from_Ms_A_K(me.Ms(1e5), me.A(1e-12), me.Ku(1e6))
+    ExtrinsicProperties(Hc=..., Mr=..., BHmax=...)
```

Comment:
We should add values to the output, now that we can do predictions, maybe adapt the input values and if possible use `# doctest: +NUMBER` for the floating-point comparison, not sure if it can be made work with the ExtrinsicProperties object. (https://docs.pytest.org/en/7.1.x/how-to/doctest.html#using-doctest-options)